syncd/CommandLineOptionsParser.cpp

```diff
@@ -164,6 +169,8 @@ void CommandLineOptionsParser::printUsage()
     std::cout << "        Enable consisteny check DB vs ASIC after comparison logic" << std::endl;
     std::cout << "    -s --syncMode" << std::endl;
     std::cout << "        Enable synchronous mode" << std::endl;
+    std::cout << "    -z --zmqSyncMode" << std::endl;
```

Comment:
>zmqSyncMode [](start = 27, length = 11)

How about:
-s --syncMode
-z --zmqMode
Throw errors if there are unsupported combination.

Reply:
you can specify -s -z, zmq forces -s internelly

Reply:
`-s -z` and `-z` seem to behave identically. I am thinking if it is more clear to do something like a) `-s -z`: log a notice that it is doing zmq sync mode; b) `-z`: log a warning that there is no `-s` given but we are forcing sync mode? Not completely sure if having an overlapped configuration between zmq sync mode and redis sync mode is good. Maybe we can make `-s` and `-z` mutually exclusive?

Reply:
so there is common part in syncd, when synchronous mode is enabled, common parts are executed, and this does not depend on whether is redis syncd mode or zmq sync mode, just need to be enabled in case of any synchronization
originally i was planning do to sync mode as a enum, so in this case it could be like none/redis/zmq (maybe in future boost), but user would need to specify another flag with that  enum string to be parsed internally to actual enum value. this way we would have only 1 possible way to enable single mode. but this would break backward compatibility with flag "-s" that we have right now

since at the syncd design stage there was no plan to add multiple modes since redis was acting as pipeline, there was no issue there. so currently i decided that when zmq sync mode will be set (-z), it will just override "-s" parameter everywhere (in OA/saiplayer) (and syncd synmode is enabled by -m param) so there will be no issue with possibly specifying 2 of them

but if you want, i can convert that param to actual enum value for potential future expansion. Also note that zmq mode requires some extra parameters, which at this point are just set to some hard coded values, and actual values could be changed via context_config.json

Reply:
so im actually thinking about 2 solusions here
* we have "-s" flag, that enables/disables synchronous mode (since there can be only 2 modes async/sync) and then we have one enum communication_mode (also 1 redis atribute) which specifies what mode is is (redis_async, redis_sync, zmq_sync) and this mode could be specified by -z param like "-s -z zmq_sync"
* we have "-s" flag same as above, and 2 enums async_mode_type (redis), and sync_mode_type (redis,zmq, possibly boost later) then default values could be specified for both via 2 extra params from command lines, for example "-s -z zmq" or "-Z redis", one of the params dont need to be implemented now since we have only 1 async mode for now 

both solutions have pros and cons, for first one, we only have 1 enum, but we need to deal with incompatible -s param (if we want to still keep "-s" param) and do error when "-s -Z redis" is specified, for second we keep -s as selector for sync/async mode, and we can specify "-s Z redis -z zmq" and it will be fine,  downside is that we need 2 enums to handle

any thoughts on that ?

Reply:
I got a little confused between `-z` and `-Z`. Do they mean communication mode for sync_mode and async_mode, respectively? If so, why is `-s -Z redis` a problem for the first option (`-Z` should not be valid anyway)? Shouldn't `-z zmq_sync` without `-s` be an error?

Reply:
Personally speaking, I would like one communication mode better since the redundant `-Z redis` in `-s -Z redis -z zmq` looks a little confusing to me. But either way looks OK to me.

Reply:
addressed, added communication mode enum as single param